src/components/WeatherBox/index.js

```diff
@@ -0,0 +1,73 @@
+import React from "react";
+import wind from "../../assets/wind.svg";
+import humidity from "../../assets/humidity.svg";
+import "./styles.scss";
+import { connect } from "react-redux";
+import { useHistory } from "react-router-dom";
+const WeatherBox = ({ weatherData, loading, error }) => {
+    const history = useHistory()
+    if(error) {
+        history.push("/error")
+    }
+  if (JSON.stringify(weatherData) === "{}" || loading) {
```

Comment:
We could have checked if object is empty or not instead of stringfying the data and comparing